File: app/src/main/java/org/ole/planet/myplanet/model/RealmMessage.java
package org.ole.planet.myplanet.model;

import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;

import io.realm.Realm;
import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class RealmMessage extends RealmObject {
    @PrimaryKey
    String id;
    private String message;

    private String time;

    private String user;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getTime() {
        return time;
    }

    public void setTime(String time) {
        this.time = time;
    }

    public String getUser() {
        return user;
    }

    public void setUser(String user) {
        this.user = user;
    }

    public static JsonElement serialize(RealmList<RealmMessage> messages) {
        JsonArray array = new JsonArray();
        for (RealmMessage ms : messages) {
            JsonObject object = new JsonObject();
            object.addProperty("user",ms.getUser());
            object.addProperty("time",ms.getTime());
            object.addProperty("message",ms.getMessage());
            array.add(object);
        }
        return array;
    }

      public static void insertFeedback(Realm mRealm, JsonObject act) {

      }
}

File: app/src/main/java/org/ole/planet/myplanet/model/RealmMessage.kt
package org.ole.planet.myplanet.model

import io.realm.kotlin.types.RealmObject
import io.realm.kotlin.types.annotations.PrimaryKey
import kotlinx.serialization.json.JsonElement
import kotlinx.serialization.json.buildJsonArray
import kotlinx.serialization.json.buildJsonObject
import kotlinx.serialization.json.put

class RealmMessage : RealmObject {
    @PrimaryKey
    var id: String? = null
    var message: String? = null
    var time: String? = null
    var user: String? = null

    companion object {
        fun serialize(messages: List<RealmMessage>): JsonElement {
            return buildJsonArray {
                messages.forEach { ms ->
                    add(buildJsonObject {
                        put("user", ms.user)
                        put("time", ms.time)
                        put("message", ms.message)
                    })
                }
            }
        }
    }
}